mypy/test/test_check.py

```diff
@@ -122,13 +122,12 @@ def run_test_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
             a = e.messages
         a = normalize_error_messages(a)
 
-        if output != a and mypy.myunit.UPDATE_TESTCASES:
+        if output != a and pytest.config.getoption('UPDATE_TESTCASES'):
             update_testcase_output(testcase, a, mypy.myunit.APPEND_TESTCASES)
 
-        assert_string_arrays_equal(
-            output, a,
+        assert output == a, \
             'Invalid type checker output ({}, line {})'.format(
-                testcase.file, testcase.line))
+                testcase.file, testcase.line)
```

Comment:
To clarify when it comes to the scope of this PR: I'd leave any improvements to the diff output (like switching to a unified diff) to a separate PR -- which could even come sooner! should be easy to do, and independent of this change -- and in this one just keep invoking the same `assert_string_arrays_equal` function to keep the change as simple as possible.